### ISSUE TITLE
[Enhancement]: Add support for the Cart & Checkout Blocks in the declaring compatibility feature

### ISSUE THREAD

**Issue by tarhi-saad** [woothemes/woocommerce#36413]
### Context

In Rukib team, we are maintaining the [WooCommerce Blocks](https://github.com/woocommerce/woocommerce-blocks) plugin! Couple of months ago we did a research spike (pdFofs-mB-p2) about managing extensions that don't support the Cart & Checkout Blocks. One of our propositions was to declare (in)compatibility with the C&C Blocks (pdFofs-mB-p2#declare-in-compatibility-with-the-cart-checkout-blocks) using the same approach in [HPOS compatibility declaration](https://github.com/woocommerce/woocommerce/wiki/High-Performance-Order-Storage-Upgrade-Recipe-Book#declaring-extension-incompatibility). During one of our pair programming and brainstorming sessions (pca54o-4NP-p2#declaring-in-compatibly-with-the-c-c-blocks) we worked in creating a prouf of concept for declaring (in)compatibility with the Cart & Checkout Blocks feature.

Declaring (in)compatibility is simple:
- Declare WC tested up to in the main plugin file(s)
- Add the correct code snippet to the main plugin file(s)

The code snippet used for declaring (in)compatibility with [HPOS](https://github.com/woocommerce/woocommerce/wiki/High-Performance-Order-Storage-Upgrade-Recipe-Book#declaring-extension-incompatibility):

```PHP
add_action( 'before_woocommerce_init', function() {
    if ( class_exists( '\Automattic\WooCommerce\Utilities\FeaturesUtil' ) ) {
        \Automattic\WooCommerce\Utilities\FeaturesUtil::declare_compatibility( 'custom_order_tables', __FILE__, true );
    }
} );
```

So, the function of interest is: `declare_compatibility( 'custom_order_tables', FILE, true );`, see [function definition](https://href.li/?https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/Utilities/FeaturesUtil.php#L45-L67):

https://github.com/woocommerce/woocommerce/blob/c3d62f113addbbf820ed5e8103a0a052092efc00/plugins/woocommerce/src/Utilities/FeaturesUtil.php#L45-L67

In theory, we want to replace the `custom_order_tables` feature_id by a C&C Blocks feature_id (e.g., `cart_checkout_blocks`). During our research, we realized that only the WooCommerce features in `WooCommerce -> Settings -> Advanced -> Features` can use the `declare_compatibility` function:

![image](https://user-images.githubusercontent.com/14235870/212312501-2ee56ff3-b78e-4f53-93bc-675947ed8071.png)

In other words, the current implementation of the `declare_compatibility` function doesn’t work for the Cart & Checkout Blocks feature.

### Solutions

I this case, we have two options:

- In Woo Blocks, create the compatibility declaration API from scratch. This, I believe, is a bad idea: we will duplicate the same feature, and we will not have a single source of truth!
- Add support for the Cart & Checkout Blocks in the declaring compatibility feature

We eventually chose to go with the second solution!
